yjit.rb

```diff
@@ -254,6 +261,33 @@ class << self
     # :stopdoc:
     private
 
+    def enable_with_custom_config(stats, log, exec_mem_size, call_threshold)
+      validate_config_params(exec_mem_size, call_threshold)
+      config_success = Primitive.rb_yjit_parse_custom_config(exec_mem_size, call_threshold)
+
+      unless config_success
+        warn 'YJIT custom configuration failed to get set. YJIT will not be enabled.'
+        return false
+      end
+
+      at_exit { print_and_dump_stats } if stats
+      Primitive.rb_yjit_enable(stats, stats != :quiet, log, log != :quiet)
+    rescue ArgumentError => e
+      warn "YJIT configuration error: #{e.message}. YJIT will not be enabled."
+      false
+    end
+
+    def validate_config_params(exec_mem_size, call_threshold)
+      if exec_mem_size && !(exec_mem_size.is_a?(Integer) && exec_mem_size.between?(1, 2048))
+        raise ArgumentError, 'Invalid exec_mem_size: must be an integer between 1 and 2048'
+      end
+
+      return unless call_threshold
+      return if call_threshold.is_a?(Integer) && call_threshold >= 1
+
+      raise ArgumentError, 'Invalid call_threshold: must be an integer >= 1'
+    end
```

Comment:
Might be a bit simpler to directly validate the values directly in the enable function.